modules/commons/src/main/frontend/src/components/DateTimeUtilities.jsx

```diff
@@ -105,16 +108,15 @@ export default class DateQuestionUtilities {
 
     // Determine the coarsest measure to truncate the input to
     const truncate = {
-      'S':'second',
       's':'minute',
       'm':'hour',
       'H':'day',
       'd':'month',
       'M':'year'
     };
-    let truncateTo;
+    let truncateTo = 'second';
     for (let [formatSpecifier, targetPrecision] of Object.entries(truncate)) {
-      if (toFormat.indexOf(formatSpecifier) < 0) {
+      if (toFormat.indexOf(formatSpecifier) < 0 && ("H" != formatSpecifier || toFormat.indexOf("h") < 0)) {
```

Comment:
This logic feels like it could use an explanation:
```suggestion
      // Check if the current specifier is missing.
      // If the current specifier is 'H', check if both 'H' and 'h' are missing
      if (toFormat.indexOf(formatSpecifier) < 0 && ("H" != formatSpecifier || toFormat.indexOf("h") < 0)) {
```